src/linked_hash_map.rs

```diff
@@ -346,6 +394,26 @@ impl<K: Hash<HmHasher> + Eq, V> LinkedHashMap<K, V> {
     }
 }
 
+impl<K, V> Index<K> for LinkedHashMap<K, V>
+    where K: Hash<HmHasher> + Eq
+{
+    type Output = V;
+
+    fn index<'a>(&'a self, index: &K) -> &'a V {
+        self.get(index).expect("not entry found for key")
+    }
+}
+
+impl<K, V> IndexMut<K> for LinkedHashMap<K, V>
+    where K: Hash<HmHasher> + Eq
+{
+    type Output = V;
+
+    fn index_mut<'a>(&'a mut self, index: &K) -> &'a mut V {
```

Comment:
@stepancheg the right thing to do here is to elide the lifetime, it's probably only in the std HashMap because of legacy reasons since elision did not exist in the past.